packages/builders/src/components/Component.ts

```diff
@@ -8,17 +14,15 @@ export abstract class Component<
 	DataType extends Partial<APIBaseComponent<ComponentType>> & {
 		type: ComponentType;
 	} = APIBaseComponent<ComponentType>,
-> implements JSONEncodable<APIMessageComponent>
+> implements JSONEncodable<APIMessageComponent>, Equatable<Component | APIActionRowComponentTypes>
 {
 	/**
 	 * The API data associated with this component
 	 */
 	protected readonly data: DataType;
 
-	/**
-	 * Converts this component to an API-compatible JSON object
-	 */
 	public abstract toJSON(): APIMessageComponent;
+	public abstract equals(other: Component | APIActionRowComponentTypes): boolean;
```

Comment:
```suggestion

	public abstract equals(other: Component | APIActionRowComponentTypes): boolean;
```
everything else gets a line in between